pkg/cmd/build/depot.go

```diff
@@ -0,0 +1,173 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/depot/depot-go/build"
+	"github.com/depot/depot-go/machine"
+	cliv1 "github.com/depot/depot-go/proto/depot/cli/v1"
+	"github.com/moby/buildkit/client"
+	buildkitClient "github.com/moby/buildkit/client"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/types"
+	"github.com/spf13/afero"
+)
+
+const (
+	depotTokenEnvVar   = "DEPOT_TOKEN"
+	depotProjectEnvVar = "DEPOT_PROJECT"
+
+	defaultPlatform = "amd64"
+)
+
+type depotMachineConnector interface {
+	Connect(ctx context.Context) (*buildkitClient.Client, error)
+	Release() error
+}
+
+type depotBuilder struct {
+	okCtx          OktetoContextInterface
+	fs             afero.Fs
+	machine        depotMachineConnector
+	err            error
+	ioCtrl         *io.IOController
+	newDepotBuild  func(ctx context.Context, req *cliv1.CreateBuildRequest, token string) (build.Build, error)
+	acquireMachine func(ctx context.Context, buildId, token, platform string) (depotMachineConnector, error)
+	token          string
+	project        string
+}
+
+// isDepotEnabled returns true if depot env vars are set
+func isDepotEnabled(depotProject, depotToken string) bool {
+	return depotToken != "" && depotProject != ""
+}
+
+// newDepotBuilder creates a new instance of DepotBuilder.
+func newDepotBuilder(projectId, token string, okCtx OktetoContextInterface, ioCtrl *io.IOController) *depotBuilder {
+	return &depotBuilder{
+		ioCtrl:  ioCtrl,
+		token:   token,
+		project: projectId,
+		okCtx:   okCtx,
+		fs:      afero.NewOsFs(),
+
+		newDepotBuild: build.NewBuild,
+		acquireMachine: func(ctx context.Context, buildId, token, platform string) (depotMachineConnector, error) {
+			return machine.Acquire(ctx, buildId, token, platform)
+		},
+	}
+}
+
+// release releases the depot's machine and finishes the build.
+func (db *depotBuilder) release(build build.Build) {
+	build.Finish(db.err)
+
+	err := db.machine.Release()
+	if err != nil {
+		db.ioCtrl.Logger().Infof("failed to release depot's machine: %s", err)
+	}
+}
+
+type runAndHandleBuildFn func(ctx context.Context, c *client.Client, opt *client.SolveOpt, buildOptions *types.BuildOptions, okCtx OktetoContextInterface, ioCtrl *io.IOController) error
+
+func (db *depotBuilder) Run(ctx context.Context, buildOptions *types.BuildOptions, run runAndHandleBuildFn) error {
+	db.ioCtrl.Logger().Info("building your image on depot's machine")
+
+	// Register a new build.
+	req := &cliv1.CreateBuildRequest{
+		ProjectId: db.project,
+		Options: []*cliv1.BuildOptions{
+			{
+				Command: cliv1.Command_COMMAND_BUILD,
+				Tags:    []string{buildOptions.Tag},
+				Push:    true,
+				Load:    true,
+			},
+		},
+	}
+
+	build, err := db.newDepotBuild(ctx, req, db.token)
+	if err != nil {
+		return err
```

Comment:
What do you think if we wrap this error with the mention of `depot`? 

Because now the error for wrong credentials looks like this:

```
 x  Error building service 'frontend': unauthenticated: Invalid token
```

If I don't know that my okteto cli is using Depot, it may look like an invalid okteto token.

```suggestion
		return fmt.Errorf("depot build failed: %w", err)
```